File: ts/wam/5462.ts
export const WamSmbUserJourney = 5462 as const

export interface SmbUserJourneyEvent {
  /** field 1, wire `action_type` */
  actionType?: CHATFILTERACTIONTYPES
  /** field 2, wire `contact_is_saved` */
  contactIsSaved?: boolean
  /** field 3, wire `entry_point` */
  entryPoint?: ENTRYPOINT
  /** field 4, wire `extra_attributes` */
  extraAttributes?: string
  /** field 5, wire `opposite_platform` */
  oppositePlatform?: OPPOSITEPLATFORMENUM
  /** field 7, wire `seq_id` */
  seqId?: number
  /** field 8, wire `smb_feature_name` */
  smbFeatureName?: SMBFEATURENAMEENUM
  /** field 9, wire `smb_user_session_id` */
  smbUserSessionId?: string
  /** field 10, wire `surface` */
  surface?: SURFACETYPE
  /** field 11, wire `user_action_target` */
  userActionTarget?: string
  /** field 12, wire `recipient_size` */
  recipientSize?: number
  /** field 13, wire `smb_user_action_type` */
  smbUserActionType?: SMBUSERACTIONTYPEENUM
  /** field 14, wire `prev_surface` */
  prevSurface?: SURFACETYPE
  /** field 18, wire `entry_point_details` */
  entryPointDetails?: string
  /** field 19, wire `has_catalog` */
  hasCatalog?: boolean
  /** field 20, wire `is_coex_account` */
  isCoexAccount?: boolean
  /** field 21, wire `is_mv_subscriber` */
  isMvSubscriber?: boolean
  /** field 22, wire `bb_tier` */
  bbTier?: BBTIERTYPE
}

File: go/wam/5462.go
const WamSmbUserJourney = 5462 // channel: regular

type SmbUserJourneyEvent struct {
	ActionType CHATFILTERACTIONTYPES `wam:"action_type"` // field 1
	ContactIsSaved bool `wam:"contact_is_saved"` // field 2
	EntryPoint ENTRYPOINT `wam:"entry_point"` // field 3
	ExtraAttributes string `wam:"extra_attributes"` // field 4
	OppositePlatform OPPOSITEPLATFORMENUM `wam:"opposite_platform"` // field 5
	SeqId int64 `wam:"seq_id"` // field 7
	SmbFeatureName SMBFEATURENAMEENUM `wam:"smb_feature_name"` // field 8
	SmbUserSessionId string `wam:"smb_user_session_id"` // field 9
	Surface SURFACETYPE `wam:"surface"` // field 10
	UserActionTarget string `wam:"user_action_target"` // field 11
	RecipientSize int64 `wam:"recipient_size"` // field 12
	SmbUserActionType SMBUSERACTIONTYPEENUM `wam:"smb_user_action_type"` // field 13
	PrevSurface SURFACETYPE `wam:"prev_surface"` // field 14
	EntryPointDetails string `wam:"entry_point_details"` // field 18
	HasCatalog bool `wam:"has_catalog"` // field 19
	IsCoexAccount bool `wam:"is_coex_account"` // field 20
	IsMvSubscriber bool `wam:"is_mv_subscriber"` // field 21
	BbTier BBTIERTYPE `wam:"bb_tier"` // field 22
}